pkg/etag_cache/etag_cache.go

```diff
@@ -0,0 +1,51 @@
+package etag_cache
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Cache struct {
+	cache map[string]*CacheEntry
+	lock  sync.RWMutex
+}
+
+type CacheEntry struct {
+	Etag        string
+	Response    []byte
+	LastUpdated time.Time
+}
+
+func NewCache() *Cache {
+	return &Cache{
+		cache: map[string]*CacheEntry{},
+		lock:  sync.RWMutex{},
+	}
+}
+
+func (c *Cache) Get(key string) *CacheEntry {
+	if c == nil {
+		return nil
+	}
+	c.lock.RLock()
```

Comment:
query: What is the reasoning behind a read lock. In other words, don't we need a write lock while setting the cache?

Reply:
not relevant anymore